packages/desktop-client/src/components/budget/MobileBudgetTable.js

```diff
@@ -248,203 +268,375 @@ function BudgetCategoryPreview({ name, pending, style }) {
+  let spent = rolloverBudget.catSumAmount(category.id);
+
+  let tooltip = useTooltip();
+
+  let onTooltipClose = () => {
+    onSave?.({
+      ...category,
+      name: categoryName,
+    });
+    tooltip.close();
+  };
+
+  let onMenuSelect = type => {
+    tooltip.close();
+    switch (type) {
+      case 'toggle-visibility':
+        setIsHidden(!isHidden);
+        onSave?.({
+          ...category,
+          hidden: !isHidden,
+        });
+        break;
+      case 'delete':
+        onDelete?.(category.id);
+        break;
+      default:
+        throw new Error(`Unrecognized category menu type: ${type}`);
+    }
+  };
+
+  let content = (
+    <ListItem
+      style={{
+        backgroundColor: editing ? theme.altTableTextEditing : 'transparent',
+        borderBottomWidth: 0,
+        borderTopWidth: index > 0 ? 1 : 0,
+        opacity: isHidden ? 0.5 : undefined,
+        ...style,
+      }}
+      data-testid="row"
+    >
+      <View style={{ flex: 1 }}>
+        <Text
           style={{
-            alignItems: 'center',
-            flexDirection: 'row',
-            opacity: this.opacity,
+            ...styles.smallText,
+            ...styles.underlinedText,
+            whiteSpace: 'nowrap',
+            overflow: 'hidden',
+            textOverflow: 'ellipsis',
           }}
+          data-testid="category-name"
+          {...tooltip.getOpenEvents()}
         >
-          {show3Cols || showBudgetedCol ? (
-            <BudgetCell
-              name="budgeted"
-              binding={budgeted}
-              editing={editing}
-              style={{ width: 90 }}
-              textStyle={{ ...styles.smallText, textAlign: 'right' }}
-              categoryId={category.id}
-              month={month}
-              onBudgetAction={onBudgetAction}
-            />
-          ) : null}
-          {show3Cols || !showBudgetedCol ? (
-            <CellValue
-              name="spent"
-              binding={spent}
-              style={{
-                ...styles.smallText,
-                width: 90,
-                textAlign: 'right',
-              }}
-              type="financial"
-            />
-          ) : null}
+          {category.name}
+          {tooltip.isOpen && (
+            <Tooltip
+              position="bottom-left"
+              width={300}
+              offset={5}
+              style={{ padding: 0 }}
+              onClose={onTooltipClose}
+            >
+              <View style={{ flexDirection: 'row' }}>
+                <InputWithContent
+                  style={{ flex: 1, margin: '10px 0px 10px 10px' }}
+                  leftContent={
+                    <Button type="bare" disabled>
+                      <EditPencil width={9} height={9} />
+                    </Button>
+                  }
+                  placeholder="Category Name"
+                  value={categoryName}
+                  onUpdate={name =>
+                    setCategoryName(name ? name : category.name)
```

Comment:
This is buggy: it prevents the user from clearing the category name field. Instead of checking whether it is empty here, we should check when the user actually submits. Only if the category name is empty *when they submit*, then the expected behaviour is to restore the original name.

This issue is also present on line 942.

Reply:
Updated in latest commit :)